packages/core/test/polygon/e2e.js

```diff
@@ -0,0 +1,176 @@
+const { ZERO_ADDRESS, interfaceName, RegistryRolesEnum, didContractThrow } = require("@uma/common");
+const { assert } = require("chai");
+const TruffleAssert = require("truffle-assertions");
+
+const { utf8ToHex, toWei, hexToUtf8 } = web3.utils;
+
+const StateSync = artifacts.require("StateSyncMock");
+const FxChild = artifacts.require("FxChildMock");
+const FxRoot = artifacts.require("FxRootMock");
+const OracleChildTunnel = artifacts.require("OracleChildTunnel");
+const OracleRootTunnel = artifacts.require("OracleRootTunnelMock");
+const Finder = artifacts.require("Finder");
+const Registry = artifacts.require("Registry");
+const MockOracle = artifacts.require("MockOracleAncillary");
+const IdentifierWhitelist = artifacts.require("IdentifierWhitelist");
+
+contract("Polygon <> Ethereum Tunnel: End-to-End Test", async (accounts) => {
```

Comment:
@UMAprotocol/eng this is the key unit test to understand integration between Polygon <> Ethereum tunnels